subsys/bluetooth/mesh/transport.c

```diff
@@ -873,8 +873,10 @@ static int ctl_recv(struct bt_mesh_net_rx *rx, u8_t hdr,
 		return trans_heartbeat(rx, buf);
 	}
 
-	/* Only acks and heartbeats may need processing without local_match */
-	if (!rx->local_match) {
+	/* Only acks and heartbeats may need processing without local_match
+	 * Avoid Processing caused by local network (LPN send REQ-->FRIEND)
+	 */
+	if (!rx->local_match || bt_mesh_elem_find(rx->ctx.addr)) {
```

Comment:
I think this is a too broad rule which could introduce bugs in the future if/when new control codes are added, i.e. when it might be perfectly valid to send those control codes to the local node. I think I'd feel better handling this on a per-opcode basis, i.e. in this case check for local source address in the Friend Request handler.